python/triqs/gf/gf.py

```diff
@@ -530,11 +530,21 @@ def _combine_mesh_mul(l, r):
     def __mul__(self,y):
         if isinstance(y, Gf):
             # make a copy, but special treatment of the mesh in the Imtime case.
-            c = Gf(mesh = Gf._combine_mesh_mul(self._mesh, y.mesh),
-                   data = self._data.copy(), 
-                   indices = self._indices.copy(), 
-                   name = self.name)
-            c.__imul__impl(y)
+            if self.target_rank == 2 and y.target_rank == 2 and self.target_shape != y.target_shape:
+                c = Gf(mesh=Gf._combine_mesh_mul(self._mesh, y.mesh),
+                       target_shape=[self.target_shape[0], y.target_shape[1]])
+                rh = lambda d : np.reshape(d, (reduce(mul, d.shape[:c.rank]),) + (d.shape[c.rank:]))
+                d_self =  rh(self.data)
+                d_y =  rh(y.data)
+                d_c = rh(c.data)
+                for n in range (d_self.shape[0]):
+                    d_c[n] = np.dot(d_self[n], d_y[n])
```

Comment:
Why all the reshapes? Shouldn't this just be
```suggestion
                np.dot(self.data, y.data, out=c.data)
```
c.f. https://numpy.org/doc/stable/reference/generated/numpy.dot.html
Quote:

> `numpy.dot(a, b, out=None)`
> If a is an N-D array and b is an M-D array (where M>=2), it is a sum product over the last axis of a and the second-to-last axis of b:
> `dot(a, b)[i,j,k,m] = sum(a[i,j,:] * b[k,:,m])`